src/main/java/com/williamfiset/algorithms/datastructures/stack/impl/ArrayStack.java

```diff
@@ -0,0 +1,62 @@
+package com.williamfiset.algorithms.datastructures.stack.impl;
+
+import com.williamfiset.algorithms.datastructures.stack.Stack;
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
+/** @author liujingkun */
+public class ArrayStack<T> implements Stack<T> {
+  private int capacity;
+  private Object[] data;
+  private int size;
+
+  public ArrayStack() {
+    // the factor of 10 is the magic of speed!!
+    capacity = 10;
+    size = 0;
+    data = new Object[capacity];
+  }
+
+  @Override
+  public int size() {
+    return size;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return 0 == size;
+  }
+
+  @Override
+  public void push(T elem) {
+    if (size == capacity) {
+      adjustCap();
+    }
+    data[size++] = elem;
+  }
+
+  // adjust the capacity to store more element
+  private void adjustCap() {
+    if (capacity == Integer.MAX_VALUE) {
```

Comment:
This line will never execute is capacity starts at 10. The capacity will overflow before this line is ever triggered. I suggest just removing it?

Reply:
the capacity starts at 16 and will not overflow